samples/sample_encode/src/sample_encode.cpp

```diff
@@ -1648,6 +1648,42 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
     }
 #endif
 
+    mfxU16 mfxU16Limit = std::numeric_limits<mfxU16>::max();
+    if ((pParams->MaxKbps && pParams->MaxKbps > mfxU16Limit) ||
+        (pParams->nBitRate && pParams->nBitRate > mfxU16Limit) ||
+        (pParams->InitialDelayInKB && pParams->InitialDelayInKB > mfxU16Limit) ||
+        (pParams->BufferSizeInKB && pParams->BufferSizeInKB > mfxU16Limit))
+    {
+        mfxU32 maxVal = std::max<mfxU32>({pParams->MaxKbps, pParams->nBitRate, pParams->InitialDelayInKB, pParams->BufferSizeInKB});
+        mfxU32 newMaxKb = maxVal;
+        mfxU32 newBitRateMultiplier = 1;
+        while (newMaxKb > mfxU16Limit)
+        {
+            newMaxKb = maxVal;
+            newBitRateMultiplier += 1;
+            newMaxKb /= newBitRateMultiplier; 
+        }
```

Comment:
why not to just:
newBitRateMultiplier = std::ceil(static_cast<float>(newMaxKb) / newBitRateMultiplier) ;

Reply:
changed